### ISSUE TITLE
App freezes when started in server mode with a server already running, then changing to client mode

### ISSUE THREAD

**Issue by SamCarlberg** [wpilibsuite/OutlineViewer#69]
Steps to reproduce:

1. Start an instance in server mode
2. Start another instance in server mode (note this should fail and the indicator should say "No connection to localhost", probably a bug in ntcore not updating the network mode)
3. Change the second instance to client mode. It will immediately freeze.

**Comment by AustinShalit** [comment]
I followed the steps to reproduce but could not get it to lock up.  Can you provide a stack trace of where it is locking up for you?

**Comment by SamCarlberg** [comment]
There's no exception, and no stack trace, but [here's a thread dump](https://gist.github.com/SamCarlberg/961eaff4a345e730fdeda1a3ba758cef)

**Comment by SamCarlberg** [comment]
Put it through a debugger, `NetworkTableInstance::stopClient` is hanging for some reason

Tagging @PeterJohnson since this appears to be an ntcore issue

**Comment by PeterJohnson** [comment]
What platform?  I can't reproduce on Linux.  The only potential issue I see is that the second server won't start up properly due to the address already being in use, and potentially then the stop doesn't work because it's not started up all the way?

**Comment by SamCarlberg** [comment]
Windows 10 home 64-bit. What's happening is `NetworkTableUtilities.shutdown()` is called and hangs on this line:

```java
public static void shutdown() {
  networkTableInstance.stopDSClient();
  networkTableInstance.stopClient(); // <--- hangs here and does not return
  networkTableInstance.stopServer();

  networkTableInstance.deleteAllEntries();
}
```

Which is _before_ `stopServer()` is called. So `stopClient()` is hanging for some reason

**Comment by PeterJohnson** [comment]
stopClient and stopServer do the exact same thing under the hood.

**Comment by PeterJohnson** [comment]
Do you see any console messages regarding "bind() to port failed"?

**Comment by SamCarlberg** [comment]
Nope. And the connection indicator shows that a server is running, or at least ntcore _thinks_ it's running

**Comment by PeterJohnson** [comment]
I believe this is a wpiutil TCPAcceptor issue specific to windows.  See wpilibsuite/wpiutil#44.
